packages/ng/dialog/dialog-header/dialog-header.component.ts

```diff
@@ -0,0 +1,49 @@
+import { NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { ButtonComponent } from '@lucca-front/ng/button';
+import { IconComponent } from '@lucca-front/ng/icon';
+import { LuDialogRef } from '../model';
+import { CdkDialogContainer } from '@angular/cdk/dialog';
+import { LU_DIALOG_HEADER_TRANSLATIONS } from './dialog-header.translate';
+import { getIntl } from '@lucca-front/ng/core';
+
+let nextId = 0;
+
+@Component({
+	selector: 'lu-dialog-header',
+	standalone: true,
+	imports: [IconComponent, ButtonComponent, NgIf],
+	templateUrl: './dialog-header.component.html',
+	styleUrl: './dialog-header.component.scss',
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: 'dialog-inside-header',
+		role: 'header',
+	},
+})
+export class DialogHeaderComponent implements OnInit {
+	#ref = inject(LuDialogRef);
+
+	intl = getIntl(LU_DIALOG_HEADER_TRANSLATIONS);
+
+	@Input('id')
+	id: string;
+
+	dismissible = this.#ref.config.dismissible !== false;
+
+	close(): void {
+		this.#ref.dismiss();
+	}
+
+	ngOnInit(): void {
+		// Using setTimeout here to make sure this will be handled in the next Cd cycle, not the current one.
+		setTimeout(() => {
```

Comment:
Is Angular missing a lifecycle hook for this use case ? 🤯

Reply:
It's more that if we change anything at this exact time, we'll get an error due to expression changed while it was checked, which could result in unapplied changes in prod environment.